sdk/storage/azure-storage-file/tests/test_file_client.py

```diff
@@ -244,6 +244,78 @@ def test_create_service_with_connection_string_succeeds_if_secondary_with_primar
             self.assertEqual(service.primary_hostname, 'www.mydomain.com')
             self.assertEqual(service.secondary_hostname, 'www-sec.mydomain.com')
 
+    def test_create_service_with_custom_account_endpoint_path(self):
+        custom_account_url = "http://local-machine:11002/custom/account/path/" + self.sas_token
+        for service_type in SERVICES.items():
+            conn_string = 'DefaultEndpointsProtocol=http;AccountName={};AccountKey={};FileEndpoint={};'.format(
+                self.account_name, self.account_key, custom_account_url)
+
+            # Act
+            service = service_type[0].from_connection_string(
+                conn_string, share_name="foo", directory_path="bar", file_path="baz")
+
+            # Assert
+            self.assertEqual(service.account_name, self.account_name)
+            self.assertEqual(service.credential.account_name, self.account_name)
+            self.assertEqual(service.credential.account_key, self.account_key)
+            self.assertEqual(service.primary_hostname, 'local-machine:11002/custom/account/path')
+        
+        service = FileServiceClient(account_url=custom_account_url)
+        self.assertEqual(service.account_name, "local-machine:11002")
+        self.assertEqual(service.credential, None)
+        self.assertEqual(service.primary_hostname, 'local-machine:11002/custom/account/path')
+        self.assertTrue(service.url.startswith('http://local-machine:11002/custom/account/path/?'))
+
+        service = ShareClient(account_url=custom_account_url, share_name="foo", snapshot="snap")
+        self.assertEqual(service.account_name, "local-machine:11002")
+        self.assertEqual(service.share_name, "foo")
+        self.assertEqual(service.snapshot, "snap")
+        self.assertEqual(service.credential, None)
+        self.assertEqual(service.primary_hostname, 'local-machine:11002/custom/account/path')
+        self.assertTrue(service.url.startswith('http://local-machine:11002/custom/account/path/foo?sharesnapshot=snap&'))
+
+        service = DirectoryClient(account_url=custom_account_url, share_name='foo', directory_path="bar/baz", snapshot="snap")
+        self.assertEqual(service.account_name, "local-machine:11002")
+        self.assertEqual(service.share_name, "foo")
+        self.assertEqual(service.directory_path, "bar/baz")
+        self.assertEqual(service.snapshot, "snap")
+        self.assertEqual(service.credential, None)
+        self.assertEqual(service.primary_hostname, 'local-machine:11002/custom/account/path')
+        # TODO: Should directory path be URL-encoded here?
+        self.assertTrue(service.url.startswith('http://local-machine:11002/custom/account/path/foo/bar%2Fbaz?sharesnapshot=snap&'))
```

Comment:
Hi @xiafu-msft 
As you can see here, the `DirectoryClient` is url-encoding the path "bar/baz" to "bar%2Fbaz". Please confirm that this behaviour is accurate :)

Reply:
Yeah I think it is correct, however if you use `http://local-machine:11002/custom/account/path/foo/bar/baz?` it will also be correct, because (I think) service will do 'unquote' on their side once they receive the request the process the directory. So 'bar%2Fbaz' will be unquoted to 'bar/baz' by service, and baz is put under bar as a subdirectory.

Reply:
Thanks for the confirmation :)